q.js

```diff
@@ -1540,6 +1540,60 @@ Promise.prototype.all = function () {
 };
 
 /**
+ * Returns the first resolved promise of an array. Prior rejected promises are
+ * ignored.  Rejects only if all promises are rejected.
+ * @param {Array*} an array containing values or promises for values
+ * @returns a promise fulfilled with the value of the first resolved promise,
+ * or a rejected promise if all promises are rejected.
+ */
+Q.any = any;
+
+function any(promises) {
+    if (promises.length === 0) {
+        return Q.resolve();
+    }
+
+    var deferred = Q.defer();
+    var promisesCount = 0;
+    array_reduce(promises, function(prev, current, index) {
+        var promise = promises[index];
+        /* Let's define that false could be a value to resolve with, so let's not
+         * if(!promise)
+         */
```

Comment:
// comment would suffice here.
But I think I disagree. Even `undefined` is a valid, and in fact common, promise resolution. Let’s not filter any value.